src/meta/processors/admin/AdminClient.cpp

```diff
@@ -567,7 +567,31 @@ folly::Future<Status> AdminClient::getLeaderDist(HostLeaderMap* result) {
     return future;
 }
 
+folly::Future<StatusOr<std::unordered_map<HostAddr, std::vector<PartitionID>>>>
+AdminClient::getLeaderDist(GraphSpaceID space) {
```

Comment:
You could directly read leader from meta now.

Reply:
This could avoid some unnecessary RPC call